README.md

```diff
@@ -91,14 +91,35 @@ Results will be added to different tagged releases along with the specific confi
 
 ## Testing
 
-Unit tests can be written for the _reference system_ to check if all nodes, topics and other requirements are met before accepting test results.
+Tests can be written for the _reference system_ to check the platform as well as if all nodes, topics and other requirements are met before accepting test results.
 
 Tests are provided to automatically generate results for you by running `colcon test` on a supported platform above.
 
-To run the test, simply run the following command from your workspace:
+### Dependencies
+
+Before running the tests there are a few prerequisites to complete:
+
+- install LTTNG and `ros2_tracing` [following the instructions in `ros2_tracing`](https://gitlab.com/ros-tracing/ros2_tracing#building)
+   - _Note:_ if you are setting up [ a realtime linux kernel for a raspberry pi using this docker file](https://github.com/ros-realtime/rt-kernel-docker-builder#raspberry-pi-4-rt-linux-kernel), it should [already include LTTNG](https://github.com/ros-realtime/rt-kernel-docker-builder/pull/18)
+- install dependencies using the following command from the `colcon_ws` directory:
+    - `rosdep install --from-paths src --ignore-src -y`
+
+Once the above steps are complete you sould be ready to run the tests and generate some results.
+
+Source your ROS distribution as well as your `ros2_tracing` overlay, compile this repository using the proper CMake arguments and generate some test results:
 
 ```
-colcon build --packages-select autoware_reference_system --cmake-args -DRUN_BENCHMARK=TRUE -DTEST_PLATFORM=TRUE
+# source your ROS distribution
+source /opt/ros/galactic/setup.bash
+# source your ros2_tracing overlay
+source /path/to/ros2_tracing_ws/install/setup.bash
+
+# cd to your colcon_ws with this repo inside
+cd /path/to/colcon_ws
+# build packages with benchmark tests enabled
+colcon build --cmake-args -DRUN_BENCHMARK=TRUE -DTEST_PLATFORM=TRUE
+# run tests, generate traces and reports
+colcon test
```

Comment:
```suggestion
# Terminal 1
mkdir -p ref_sys_ws/src
cd ref_sys_ws/src
# Source your ROS distribution
source /opt/ros/galactic/setup.bash
# Clone repos
git clone https://github.com/ros-realtime/reference-system.git
git clone https://gitlab.com/ros-tracing/ros2_tracing.git
# Build packages with benchmark tests enabled
colcon build --cmake-args -DRUN_BENCHMARK=TRUE -DTEST_PLATFORM=TRUE
```

```
# Terminal 2
# Source workspace
source install/setup.bash
# Run tests, generate traces and reports
colcon test --packages-select $(colcon list -n --base-paths src/reference-system/)
```

this works for me (userspace traces have `ros2:*` events), but only for `profile-autoware_default_staticsinglethreaded_rmw_cyclonedds_cpp-rmw_cyclonedds_cpp-5s`, and I have no idea why. Any clue?

By the way, the timeout value for `generate_traces` (currently 5 seconds) doesn't seem to be actually controlling the length/time of the test execution

Reply:
@christophebedard so I'm getting `ros2:*` events for all traces, however whats missing in the traces that the `tracetools process` is complaining about are the `ros2:rcl_init` events. do you know why the traces would be missing those events?

Reply:
I can't think of a reason why the traces would have `ros2:*` userspace events but no `ros2:rcl_init` events. Can you upload the traces again?

Reply:
> By the way, the timeout value for generate_traces (currently 5 seconds) doesn't seem to be actually controlling the length/time of the test execution

this was the problem. the traces being generated were way too short so nothing was ever registered. I just fixed it to now run the traces for the set time and now the `process` call works just fine.